### ISSUE TITLE
Vehicle object model: Implement Estimator status messages and visual indicators

### ISSUE THREAD

**Issue by jcooper21** [mavlink/qgroundcontrol#2833]
A couple of great features to see in Qground control would be vibe monitoring, EKF monitoring, second GPS sat info, Wind speed & direction and current draw.

**Comment by DonLakeFlyer** [comment]
I tried getting current from the ardupilot BATTERY2 message but that message didn't seem to be sent. Is there some other way to get current? On GPS sat info, what sort of info do you want (or point me to what in MP already shows that)?

**Comment by jcooper21** [comment]
Mission planner shows the sat count number of the second GPS when you click the second mouse button on the OSD and select sat 2 count. That is also how wind speed and direction are displayed. I am not sure how to get the battery current info.

**Comment by DonLakeFlyer** [comment]
Ok. I'll add those as properties to the configurable instrument panel section. You know about that right? Does mission planner, show current somewhere. If it can, then QGC can.

**Comment by jcooper21** [comment]
Yes. Mission planner shows current right beside where it says voltage on the OSD on the flight data screen. Just to the right of voltage.

**Comment by jcooper21** [comment]
Everything I noted is all in the flight data screen on the OSD. No I don't know about instrument panel.  I am having issues trying to change the measurement units in Windows. Is that where I can change measurement units in QGC flight data screen OSD widget?

**Comment by DonLakeFlyer** [comment]
Change of measurement units is not yet supported. If you click the instrument panel on the right you can pick additional properties to show:
![screen shot 2016-02-14 at 11 56 39 am](https://cloud.githubusercontent.com/assets/5876851/13123478/2211e720-d571-11e5-93a4-0d92b40547e5.png)

**Comment by jcooper21** [comment]
That's perfect place for the items I mentioned. Also, how come Windows has units in absolute altitude and Mac is relative, also with speed Mac is in m/s and windows is km/h. I am running APM plane 3.5

**Comment by DonLakeFlyer** [comment]
We just recently change from absolute to relative as default. If you had run a build where absolute was default (a period of a few days) it likely saved that as your setting. You'll need to set it to relative manually.

**Comment by DonLakeFlyer** [comment]
@LorenzMeier The EKF and WIND ArduPilot dialect messages seems like they should be generic. Should they be moved to common and supported there instead? Or new messages?

**Comment by LorenzMeier** [comment]
Yes, they should move to common, and yes it needs new messages. I'll get the discussion going. You will need to support the common message and the APM message in parallel for some time.

**Comment by DonLakeFlyer** [comment]
If we get the new messages in, then I can generate the new messages from the ArduPilot dialect messages using my mavlink magic firmware plugin converter thingy. For now I'll plumb the new vehicle object model in.

What about the VIBE thing? Useful to generic as well? Or firmware specific?

**Comment by LorenzMeier** [comment]
Yes, generic. Although I honestly believe it might be more efficient factoring it into the estimator status report.

**Comment by LorenzMeier** [comment]
https://github.com/mavlink/mavlink/issues/511

https://github.com/mavlink/mavlink/issues/510

**Comment by DonLakeFlyer** [comment]
MP screen shots:
![screen shot 2016-02-23 at 3 16 51 pm](https://cloud.githubusercontent.com/assets/5876851/13270085/8b4f102c-da40-11e5-8736-f97252f8d17a.png)
![screen shot 2016-02-23 at 3 17 00 pm](https://cloud.githubusercontent.com/assets/5876851/13270086/8b6fb69c-da40-11e5-8b83-fa8004e280ff.png)
